nengolib/stats/ntmdists.py

```diff
@@ -237,6 +238,77 @@ def sample(self, n, d=1, rng=np.random):
         return i4_sobol_generate(d, n, skip=0)
 
 
+def _rd_generate(n, d, seed=0.5):
+    """Generates the first ``n`` points in the ``R_d`` sequence."""
+
+    # http://extremelearning.com.au/unreasonable-effectiveness-of-quasirandom-sequences/
+    def gamma(d, n_iter=20):
+        """Newton-Raphson-Method to calculate g = phi_d."""
+        x = 1.0
+        for _ in range(n_iter):
+            x -= (x**(d + 1) - x - 1) / ((d + 1) * x**d - 1)
+        return x
+
+    g = gamma(d)
+    alpha = np.zeros(d)
+    for j in range(d):
+        alpha[j] = (1/g) ** (j + 1) % 1
+
+    z = np.zeros((n, d))
+    z[0] = (seed + alpha) % 1
+    for i in range(1, n):
+        z[i] = (z[i-1] + alpha) % 1
```

Comment:
this can be parallelized with
`z = (numpy.arange(1, n + 1)[:, numpy.newaxis] * alpha + seed) % 1`